### PR TITLE
[784] | API to add hospitals

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

File: src/main/java/in/projecteka/devservice/bridge/model/BridgeServiceRequest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+@Value
+@Builder
+public class BridgeServiceRequest {
+    @NotBlank(message = "hospital id can't be blank")
+    String id;
+    @NotBlank(message = "hospital name can't be blank")
+    String name;
+    @NotNull(message = "hospital name alias can't be empty")
+    List<String> alias;
```
Comment (MounicaSagi) [inline src/main/java/in/projecteka/devservice/bridge/model/BridgeServiceRequest.java:18]: Service can be either  hospital/lab. May be you can rename hospital with service in the messages

File: src/test/java/in/projecteka/devservice/bridge/BridgeTest.java [hunk 2]
```diff
@@ -54,4 +64,35 @@ void shouldUpdateBridgeUrl() {
                 bridgeRequest.getUrl(),
                 session.getTokenType() + " " + session.getAccessToken());
     }
+
+    @Test
+    void shouldUpdateBridgeServiceEntry() {
```
Comment (MounicaSagi) [inline src/test/java/in/projecteka/devservice/bridge/BridgeTest.java]: You can change test name to shouldPopulateBridgeServiceEntry or shouldUpsertBridgeServiceEntry as it is doing both add & update

File: src/test/java/in/projecteka/devservice/bridge/BridgeUserJourneyTest.java [hunk 3]
```diff
@@ -67,4 +81,44 @@ void shouldUpdateBridgeUrl() {
                 .expectStatus()
                 .isOk();
     }
+
+    @Test
+    void shouldUpdateBridgeServiceRequest() {
```
Comment (MounicaSagi) [inline src/test/java/in/projecteka/devservice/bridge/BridgeUserJourneyTest.java]: same here.

File: src/main/java/in/projecteka/devservice/clients/ServiceAuthenticationClient.java [hunk 4]
```diff
@@ -63,6 +66,39 @@ private BridgeUpdateRequest bridgeRequestWith(String id, String url) {
                 .then();
     }
 
+    private BridgeServiceRequest bridgeServiceRequest(String id, String name, ServiceType type, boolean active) {
+        return new BridgeServiceRequest(id, name, type, active);
+    }
+
+    public Mono<Void> upsertBridgeServiceEntry(String bridgeId, String id,
```
Comment (MounicaSagi) [inline src/main/java/in/projecteka/devservice/clients/ServiceAuthenticationClient.java:83]: rename method name to populate/upsert BridgeServiceEntry as it is doing both add & update

File: src/main/java/in/projecteka/devservice/bridge/model/BridgeServiceRequest.java (hunk 1 above)
Comment (avghadsangita) [reply src/main/java/in/projecteka/devservice/bridge/model/BridgeServiceRequest.java:18]: done

File: src/test/java/in/projecteka/devservice/bridge/BridgeTest.java (hunk 2 above)
Comment (avghadsangita) [reply src/test/java/in/projecteka/devservice/bridge/BridgeTest.java]: done.changed to shouldPopulateBridgeServiceEntry

File: src/test/java/in/projecteka/devservice/bridge/BridgeUserJourneyTest.java (hunk 3 above)
Comment (avghadsangita) [reply src/test/java/in/projecteka/devservice/bridge/BridgeUserJourneyTest.java]: changed to shouldUpsertBridgeServiceRequest

File: src/main/java/in/projecteka/devservice/clients/ServiceAuthenticationClient.java (hunk 4 above)
Comment (avghadsangita) [reply src/main/java/in/projecteka/devservice/clients/ServiceAuthenticationClient.java:83]: it is upsertBridgeServiceEntry

**Review by MounicaSagi — APPROVED** [review]

File: src/main/java/in/projecteka/devservice/clients/ServiceAuthenticationClient.java [hunk 5]
```diff
@@ -63,6 +66,39 @@ private BridgeUpdateRequest bridgeRequestWith(String id, String url) {
...
+                                               String name, ServiceType type,
+                                               boolean active, Session session
+    ) {
+        return webClient.put()
+                .uri(format("/internal/bridges/%s/services", bridgeId))
+                .contentType(MediaType.APPLICATION_JSON)
+                .header(AUTHORIZATION, format("%s %s", session.getTokenType(), session.getAccessToken()))
+                .body(Mono.just(List.of(bridgeServiceRequest(id, name, type, active))), BridgeServiceRequest.class)
+                .retrieve()
+                .onStatus(httpStatus -> httpStatus.value() == HttpStatus.BAD_REQUEST.value(),
```
Comment (mddubey) [inline src/main/java/in/projecteka/devservice/clients/ServiceAuthenticationClient.java:93]: Please add logs whenever there are failures in the network calls.

**Pull request merged**